### PR TITLE
Add NewConnectionTransport from FMPURI

### PR DESCRIPTION
**Pull request by jacobhaven**
Moving this logic from https://github.com/keybase/gregor/blame/master/rpc/server/transport.go
I think it should just mirror `rpc.NewTransport`

r? @mmaxim @jzila @maxtaco

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -49,6 +49,50 @@ type ConnectionTransport interface {
...
+type connTransport struct {
+	uri             *FMPURI
+	l               LogFactory
+	wef             WrapErrorFunc
+	conn            net.Conn
+	transport       Transporter
+	stagedTransport Transporter
+}
+
+var _ ConnectionTransport = (*connTransport)(nil)
```
Comment (jzila) [inline connection.go]: I like the approach, but we have too much overloading of the term `transport` at this point. Can we maybe do a bit of a naming refactor?

File: connection.go [hunk 2]
```diff
@@ -49,6 +49,50 @@ type ConnectionTransport interface {
...
+}
+
+func (t *connTransport) Finalize() {
+	t.transport = t.stagedTransport
+	t.stagedTransport = nil
+}
+
+func (t *connTransport) Close() {
+	t.conn.Close()
+	t.transport = nil
```
Comment (jacobhaven) [inline connection.go]: I added this so that `IsConnected()` will be `false`.

**Comment by jzila** [comment]
other than naming confusion, LGTM

File: connection.go (hunk 1 above)
Comment (jacobhaven) [inline connection.go]: The API is the API. Changing it would require a bunch of changes in kbfs code that I don't know about.

Comment (jzila) [inline connection.go]: so it goes :(

**Pull request merged**